src/components/Dashboard.svelte

```diff
@@ -3,9 +3,32 @@
   import Provider from '$components/Provider.svelte';
   import DashboardHeader from '$components/DashboardHeader.svelte';
   import ChainStatus from '$components/ChainStatus.svelte';
+  import { fetchAccountsForNetwork } from '$lib/stores/accountsStore';
+  import { user } from '$lib/stores/userStore';
+  import type { ApiPromise } from '@polkadot/api';
+  import { dotApi } from '$lib/stores';
+  import { onMount } from 'svelte';
+  import type { web3Enable, web3Accounts } from '@polkadot/extension-dapp';
+
+  let thisWeb3Accounts: typeof web3Accounts;
+  let thisWeb3Enable: typeof web3Enable;
+
+  onMount(async () => {
+    const extension = await import('@polkadot/extension-dapp');
```

Comment:
I'm curious; why do we need to await importing this here, instead of just importing at the top with the rest of the imports? Are there side-effects to importing?

Reply:
You have to do it from `onMount` because `@polkadot/extension-dapp` tries to access window on the server side:

```
3:17:05 PM [vite] Error when evaluating SSR module /src/components/ProviderLogin.svelte: failed to import "@polkadot/extension-dapp"
|- ReferenceError: window is not defined
    at file:///Users/shannonwells/github/provider-dashboard/node_modules/@polkadot/extension-dapp/bundle.js:6:13
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async nodeImport (file:///Users/shannonwells/github/provider-dashboard/node_modules/vite/dist/node/chunks/dep-75f53616.js:55970:17)
    at async ssrImport (file:///Users/shannonwells/github/provider-dashboard/node_modules/vite/dist/node/chunks/dep-75f53616.js:55863:24)
    at async eval (/Users/shannonwells/github/provider-dashboard/src/components/ProviderLogin.svelte:15:32)
    at async instantiateModule (file:///Users/shannonwells/github/provider-dashboard/node_modules/vite/dist/node/chunks/dep-75f53616.js:55925:9)
```

Reply:
Ah, okay, so there are side-effects to importing (ie, code gets executed on import). Also, I didn't realize there was SSR; I assumed this was a SPA client-side only app.

Reply:
On a side note, in Control Panel we opted not to use `@polkadot/extension-dapp`, but only `@polkadot/extension-inject`, and roll our own wallet-interaction code, because their dApp library causes a sub-par UX (not as important for Provider Dashboard, probably, given the purpose & audience).